x-pack/plugins/security_solution/public/common/hooks/eql/use_eql_preview.test.ts

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Unit } from '@elastic/datemath';
+import { renderHook, act } from '@testing-library/react-hooks';
+import { of, throwError } from 'rxjs';
+import { delay } from 'rxjs/operators';
+
+import * as i18n from '../translations';
+import { EqlSearchStrategyResponse } from '../../../../../data_enhanced/common';
+import { Source } from './types';
+import { EqlSearchResponse } from '../../../../common/detection_engine/types';
+import { useKibana } from '../../../common/lib/kibana';
+import { useEqlPreview } from '.';
+import { getMockResponse } from './helpers.test';
+
+jest.mock('../../../common/lib/kibana');
+
+describe('useEqlPreview', () => {
```

Comment:
@patrykkopycinski  I was able to test this hook more thoroughly, and was going to do the same for `useMatrixHistogram` but required one small change of first declaring the variable `let subscription$` and then assigning it. Otherwise, the tests failed saying the variable was unreachable. If this seems like an ok change, happy to add similar tests to `useMatrixHistogram`.